desc/integrals/bounce_utils.py

```diff
@@ -27,36 +27,34 @@
 )
 
 
-def get_pitch_inv(min_B, max_B, num, relative_shift=1e-6):
-    """Return 1/λ values for quadrature between ``min_B`` and ``max_B``.
+def get_pitch_inv_quad(min_B, max_B, num_pitch):
+    """Return 1/λ values and weights for quadrature between ``min_B`` and ``max_B``.
 
     Parameters
     ----------
     min_B : jnp.ndarray
         Minimum |B| value.
     max_B : jnp.ndarray
         Maximum |B| value.
-    num : int
-        Number of values, not including endpoints.
-    relative_shift : float
-        Relative amount to shift maxima down and minima up to avoid floating point
-        errors in downstream routines.
+    num_pitch : int
+        Number of values.
 
     Returns
     -------
-    pitch_inv : jnp.ndarray
-        Shape (*min_B.shape, num + 2).
-        1/λ values.
+    pitch_inv, weight : (jnp.ndarray, jnp.ndarray)
+        Shape (*min_B.shape, num_pitch).
+        1/λ values and weights.
 
     """
-    # Floating point error impedes consistent detection of bounce points riding
-    # extrema. Shift values slightly to resolve this issue.
-    min_B = (1 + relative_shift) * min_B
-    max_B = (1 - relative_shift) * max_B
+    errorif(
+        num_pitch > 1e5,
+        msg=f"{num_pitch} > 1e5 not possible due to floating point errors.",
+    )
     # Samples should be uniformly spaced in |B| and not λ (GitHub issue #1228).
-    pitch_inv = jnp.moveaxis(composite_linspace(jnp.stack([min_B, max_B]), num), 0, -1)
-    assert pitch_inv.shape == (*min_B.shape, num + 2)
-    return pitch_inv
+    x, weight = chebgauss_uniform(num_pitch)
```

Comment:
isnt the end result the same as before, with linear spacing between 1/bmin 1/bmax?

Reply:
Yes; The difference is that it’s an open quadrature, which is better because gammac is infinite at max|B| so you don’t want a closed quadrature

Reply:
linear spacing between bmin and bmax, not pitch